src/app/containers/App/App.jsx

```diff
@@ -58,15 +69,20 @@ export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
         errorCode: null,
       });
 
-      route.getInitialData(location.pathname).then(data =>
-        setState(prevState => ({
-          ...prevState,
-          loading: false,
-          pageData: path(['pageData'], data),
-          status: path(['status'], data),
-          error: path(['error'], data),
-        })),
-      );
+      route
+        .getInitialData(location.pathname)
+        .then(data =>
+          setState(prevState => ({
+            ...prevState,
+            loading: false,
+            pageData: path(['pageData'], data),
+            status: path(['status'], data),
+            error: path(['error'], data),
+          })),
+        )
+        .then(() => {
+          shouldSetFocus.current = true;
+        });
```

Comment:
Yeah, I can't see a reason for the seperate promise myself?

Reply:
This is to ensure that this 
```
if (shouldSetFocus.current) {
      const contentEl = document.querySelector('h1#content');
      if (contentEl) {
        contentEl.focus();
      }
      shouldSetFocus.current = false;
    }
```
only runs after `setState` must have run. Since `setState` is asynchronous, if its added to the first promise, isn't there a chance that `shouldSetFocus.current` will be set before `setState` executes?

Reply:
To ensure that we set focus after `route.getInitialData` has resolved and `setState` has been executed. `shouldSetFocus` is attached to `state.loading` because the first `setState` call also alters the `loading` state, and we do not want to set focus for this. we want to set focus only when a new route is rendered and loading has been set to `false`.